classes/Currency.php

```diff
@@ -402,18 +402,20 @@ public static function getCurrencies($object = false, $active = true, $groupBy =
      * @param bool $active If true only active are returned
      * @param bool $groupBy Group by id_currency
      * @param bool $currentShopOnly If true returns only currencies associated to current shop
+     * @param bool $filterDeleted It true filter deleted currencies
      *
      * @return array Currency data from database
      *
      * @throws PrestaShopDatabaseException
      */
-    public static function findAll($active = true, $groupBy = false, $currentShopOnly = true)
+    public static function findAll($active = true, $groupBy = false, $currentShopOnly = true, $filterDeleted = true)
     {
         $currencies = Db::getInstance()->executeS('
             SELECT *
             FROM `' . _DB_PREFIX_ . 'currency` c
             ' . ($currentShopOnly ? Shop::addSqlAssociation('currency', 'c') : '') . '
-                WHERE `deleted` = 0' .
+                WHERE 1' .
+                ($filterDeleted ? ' AND c.`deleted` = 0' : '') .
```

Comment:
Maybe to would be nice to refactor this SQL to be built using `DbQuery` class?

Reply:
Yep great idea, I tried to make it quickly But there's a problem because of `Shop::addSqlAssociation` which simply returns the SQL condition, so it would require refacto this part as well
But I'm not sure it's worth it considering this PR scope, and it would risk adding some bugs as well This should be done in a dedicated PR with appropriate tests

Reply:
Agreed for keeping this outside of this PR scope